src/module/actor/helpers.ts

```diff
@@ -271,9 +270,9 @@ function createEnvironmentRollOptions(actor: ActorPF2e): Record<string, boolean>
             const top = region.elevation.top ?? Infinity;
             if (token.elevation < bottom || token.elevation > top) continue;
 
-            for (const behavior of region.behaviors.filter((b) => b.type === "pf2eEnvironment")) {
```

Comment:
tbh the reason I'd prefer to filter like this is because I think having separate loops for environment or terrain or whatever we do would be easier to read. The performance impact is just the creation of new arrays, since looping twice with one if per is the same performance impact as looping once with two if statements per.

Reply:
Filter is restored but I had to change the `RegionBehaviorTypeInstances` `TParent` to always have a `Scene` because apparently the token parent is narrowed to `NonNullable` somewhere. Tsc complains that `ScenePF2e` isn't `ScenePF2e | null` and I can't find a reason for that.

Reply:
I think its because until the next ts version, you have to make all filters into type guards. so `filter((b): b is whatever =>`. The next TS version will make that unnecessary.

Reply:
![image](https://github.com/foundryvtt/pf2e/assets/41452412/5afdbf7a-07eb-4207-9aa8-5732fb10b477)

This is what I get when I don't change the `RegionBehaviorTypeInstances` and I've already spent an hour before trying to figure out why that happens. 🥲

Reply:
But I agree that in 5.5 it would probably enough to filter on `b.isOfType("pf2eEnvionment")`. Can't wait for that release.

Reply:
It seems to be because RegionBehaviorTypeInstances uses ScenePF2e and not ScenePF2e | null.

Reply:
The screenshot happens with `RegionBehaviorTypeInstances<TParent extends RegionDocumentPF2e = RegionDocumentPF2e>`. Explicitly settings `RegionDocumentPF2e<ScenePF2e>` fixes it.